src/dash-playlist-loader.js

```diff
@@ -681,9 +681,10 @@ export default class DashPlaylistLoader extends EventTarget {
 
       const master = this.parseMasterXml();
       const updatedMaster = updateMaster(this.master, master);
+      const currentSidxInfo = this.media().sidx;
 
-      if (updatedMaster) {
-        const sidxKey = generateSidxKey(this.media().sidx);
+      if (updatedMaster && currentSidxInfo) {
```

Comment:
at first glance just this change makes sense but then you realize that line 717/718 should be run when we have an updatedMaster but no sidx.
Maybe we need to move the generateSidxKey inside a a currentSidxInfo check inside the updatedMaster check? If that makes sense.

Reply:
I think this change makes sense. If there's no sidx, the current master should be updated to be the `updatedMaster`

Reply:
Ah, sorry I think it would make sense if the change is to line 689/690:

``` if (currentSidxInfo && !this.sidxMapping_[sidxKey]```

we might then need to have `sidxKey` only run when there's a sidx as @gkatsev suggested.

```js
if (updatedMaster) {
  if (currentSidxInfo) {
     const sidxKey = generateSidxKey(this.media().sidx); 

     if (!this.sidxMapping_[sidxKey]) {
```

Reply:
Thanks, that make sense. I fixed it